### PR TITLE
Migrate ETL to AWS Lamba function using Serverless Framework

### PR DESCRIPTION
**Pull request by marielestavel**
```sh
# Install the Serverless Framework
$ npm install -g serverless

# Install the necessary plugins
$ serverless plugin install -n serverless-python-requirements

# Put the token received into a file called serverless.env.yml, like this
$ cat serverless.env.yml
MONGO_DB: <your_token>
POSTGRES_DB: <your_token>

# Deploy it!
$ serverless deploy
```

### CODE REVIEW

**Comment by marielestavel** [comment]
@louisguitton When I try to deploy my function by running `serverless deploy`, I get this error:
Screenshot 2023-02-19 at 16 48 52
Even though I made sure to create a virtual environment with python 3.8 using pyenv (`pyenv virtualenv 3.8 serverless`).

```
python -V
> Python 3.8.16
```

**Comment by marielestavel** [comment]
Secondly, I thought about first making sure that my script works before deploying it in order to isolate the concerns.
So I ran:
```sh
python handler.py
```

Again, I am facing an issue because it seems like `pymongo` can not connect to the server 

```
pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [Errno 61] Connection refused, Timeout: 30s, Topology Description: <TopologyDescription id: 63f24859fcb336e5aeeeb76c, topology_type: Single, servers: [<ServerDescription ('localhost', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('localhost:27017: [Errno 61] Connection refused')>]>
```

It is weird because I did think about adding my environment variables to the `serverless.env.yml` file and I am able to connect to the database with python (so my credentials are correct).

Any idea what I did wrong @louisguitton ?

File: etl-la-porte-bleue/serverless.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+service: etl-la-porte-bleue
+
+frameworkVersion: '3'
+
+provider:
+  name: aws
+  runtime: python3.8
+  environment:
+    MONGO_DB: ${file(./serverless.env.yml):MONGO_DB, ''}
+    POSTGRES_DB: ${file(./serverless.env.yml):POSTGRES_DB, ''}
```
Comment (louisguitton) [inline etl-la-porte-bleue/serverless.yml:12]: are you sure about that syntax?
from their docs, I see this https://www.serverless.com/framework/docs/providers/aws/guide/variables#referencing-environment-variables
```suggestion
    MONGO_DB: ${env:MONGO_DB, ''}
    POSTGRES_DB: ${env:POSTGRES_DB, ''}
```
for env files 
and https://www.serverless.com/framework/docs/providers/aws/guide/variables#reference-properties-in-other-files
for yaml files.
are you sure your yaml file is correct? 
```yaml
# serverless.env.yml
MONGO_DB: xxxx
POSTGRES_DB: yyyy
```
note the `:`,  not a `=` like in env files

**Review by louisguitton — COMMENTED** [review]
for question 2

File: etl-la-porte-bleue/README.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    "body": "{\"message\": \"Go Serverless v3.0! Your function executed successfully!\", \"input\": {}}"
+}
+```
+
+### Local development
+
+You can invoke your function locally by using the following command:
+
+```bash
+serverless invoke local --function hello
```
Comment (louisguitton) [inline etl-la-porte-bleue/README.md:62]: this is a great way to debug your code: 
1. try to run `python handler.py` -> this way you eliminate issues with the python code itself
2. try to run `serverless invoke local --function extract` and `serverless invoke local --function load` to check if the code works inside serverless locally
3. try to serverless deploy

**Review by louisguitton — COMMENTED** [review]
an idea to debug this

**Comment by louisguitton** [comment]
for your first question, can't tell anything wrong after a quick look at the code.
my advice is to look at the serverless documentation and to look for python + AWS examples: https://www.serverless.com/examples?prod_EXAMPLES_SEARCH_GROWTH%5BrefinementList%5D%5Bplatform%5D%5B0%5D=aws&prod_EXAMPLES_SEARCH_GROWTH%5BrefinementList%5D%5Blanguage%5D%5B0%5D=python

there is this exmaple of a pymongo serverless python project https://github.com/serverless/examples/tree/v3/aws-python-rest-api-with-pymongo

**Pull request merged**